activiti-cloud-audit-service/activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/ProcessDeletedEventConverter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.audit.jpa.converters;
+
+import java.util.List;
+import org.activiti.api.process.model.events.ProcessRuntimeEvent;
+import org.activiti.cloud.api.model.shared.events.CloudRuntimeEvent;
+import org.activiti.cloud.api.model.shared.impl.events.CloudRuntimeEventImpl;
+import org.activiti.cloud.api.process.model.events.CloudProcessDeletedEvent;
+import org.activiti.cloud.api.process.model.impl.events.CloudProcessDeletedEventImpl;
+import org.activiti.cloud.services.audit.jpa.events.AuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.events.ProcessAuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.events.ProcessDeletedAuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.repository.EventSpecificationsBuilder;
+import org.activiti.cloud.services.audit.jpa.repository.EventsRepository;
+import org.activiti.cloud.services.audit.jpa.repository.SearchOperation;
+import org.activiti.cloud.services.audit.jpa.repository.SpecSearchCriteria;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Order;
+import org.springframework.data.jpa.domain.Specification;
+
+public class ProcessDeletedEventConverter extends BaseEventToEntityConverter {
+
+    private final String PROCESS_INSTANCE_ID = "processInstanceId";
+
+    private final String TIMESTAMP = "timestamp";
+
+    private final String MISSING_PROCESS_INSTANCE = "Process Instance %s not found";
+
+    private final EventsRepository eventsRepository;
+
+    public ProcessDeletedEventConverter(EventsRepository eventsRepository, EventContextInfoAppender eventContextInfoAppender) {
+        super(eventContextInfoAppender);
+        this.eventsRepository = eventsRepository;
+    }
+
+    @Override
+    public String getSupportedEvent() {
+        return ProcessRuntimeEvent.ProcessEvents.PROCESS_DELETED.name();
+    }
+
+    @Override
+    protected ProcessDeletedAuditEventEntity createEventEntity(CloudRuntimeEvent cloudRuntimeEvent) {
+        ProcessAuditEventEntity event = findEvent(cloudRuntimeEvent.getProcessInstanceId());
```

Comment:
Is it mandatory to have other information than the process instance id set? If not, I'd avoid looking for the latest event related to this process instance, but only store the received event as it is. Then if someone is interested to know more about this process instance, they can look into the other events related to the same process instance. Does it make sense?